common/array.h

```diff
@@ -296,26 +310,32 @@ class Array {
 		return true;
 	}
 
+    /** Check if two arrays are different. */
 	bool operator!=(const Array<T> &other) const {
 		return !(*this == other);
 	}
 
+    /** Return a pointer to the underlying memory where the array begins. */
 	iterator       begin() {
 		return _storage;
 	}
 
+    /** Return a pointer to the underlying memory where the array ends. */
 	iterator       end() {
 		return _storage + _size;
 	}
 
+    /** Return a pointer to the underlying memory where the array begins. */
 	const_iterator begin() const {
 		return _storage;
 	}
 
+    /** Return a pointer to the underlying memory where the array ends. */
 	const_iterator end() const {
 		return _storage + _size;
 	}
 
+    /** Copy array data to a new place in memory. */
```

Comment:
It might copy the array data to a new place in memory, but this is a side effect, and may not happen.

I would suggest: `Reserve enough memory in the array so that it can store at least the given number of elements. The current content of the array is not modified.`